src/main/scala/com/campudus/tableaux/database/model/ServiceModel.scala

```diff
@@ -0,0 +1,197 @@
+package com.campudus.tableaux.database.model
+
+import com.campudus.tableaux.ShouldBeUniqueException
+import com.campudus.tableaux.database._
+import com.campudus.tableaux.database.domain._
+import com.campudus.tableaux.database.model.ServiceModel.ServiceId
+import com.campudus.tableaux.database.model.TableauxModel.Ordering
+import com.campudus.tableaux.helper.JsonUtils
+import com.campudus.tableaux.helper.ResultChecker._
+import org.vertx.scala.core.json.{Json, JsonArray, JsonObject}
+
+import scala.concurrent.Future
+
+object ServiceModel {
+  type ServiceId = Long
+  type Ordering = Long
+
+  def apply(connection: DatabaseConnection): ServiceModel = {
+    new ServiceModel(connection)
+  }
+}
+
+class ServiceModel(override protected[this] val connection: DatabaseConnection) extends DatabaseQuery {
+  val table: String = "system_services"
+
+  def update(
+      serviceId: ServiceId,
+      name: Option[String],
+      serviceType: Option[ServiceType],
+      ordering: Option[Ordering],
+      displayName: Option[MultiLanguageValue[String]],
+      description: Option[MultiLanguageValue[String]],
+      active: Option[Boolean],
+      config: Option[JsonObject],
+      scope: Option[JsonObject]
+  ): Future[Unit] = {
+
+    val updateParamOpts = Map(
+      "name" -> name,
+      "type" -> serviceType,
+      "ordering" -> ordering,
+      "displayName" -> displayName,
+      "description" -> description,
+      "active" -> active,
+      "config" -> config,
+      "scope" -> scope
+    )
+
+    val paramsToUpdate = updateParamOpts
+      .filter({ case (_, v) => v.isDefined })
+      .map({ case (k, v) => (k, v.orNull) })
```

Comment:
Here, just "v.get" suffices, doesn't it? Otherwise, the "a.toString" just below will fail from time to time...